app/models/concerns/article.rb

```diff
@@ -87,5 +95,13 @@ def send_to_alegre(id)
         ::Bot::Alegre.send_field_to_similarity_index(obj.project_media, field)
       end unless obj.nil?
     end
+
+    def delete_permanently(id)
+      obj = self.find_by_id(id)
+      if obj && obj.trashed
+        obj.destroy!
+        obj.claim_description.destroy! if obj.is_a?(FactCheck)
```

Comment:
Suggest to swap the lines as I prefer to destroy the object itself at the end.

Reply:
@melsawy good catch! I did that before but it had an issue because of the destroy dependency between claim and fact-check.